core/src/mender-artifact.c

```diff
@@ -211,18 +211,18 @@ mender_artifact_process_data(mender_artifact_ctx_t *ctx,
         } else if (MENDER_ARTIFACT_STREAM_STATE_PARSING_DATA == ctx->stream_state) {
 
             /* Treatment depending of the file name */
-            if (!strcmp(ctx->file.name, "version")) {
+            if (!StringEqual(ctx->file.name, "version")) {
```

Comment:
`strcmp()` will return zero (i.e., `false`) if the strings equal and non-zero otherwise (i.e., `true`). `strcmp() == 0` will do the opposite. Hence, you will need to remove the negation (i.e., `!`) here and many other places.

Reply:
Oops, went a bit too fast there